worlds/jakanddaxter/__init__.py

```diff
@@ -0,0 +1,493 @@
+        "Precursor Orbs - FC": set(Orbs.locFC_orbBundleTable.values()),
+        "Precursor Orbs - RV": set(Orbs.locRV_orbBundleTable.values()),
+        "Precursor Orbs - PB": set(Orbs.locPB_orbBundleTable.values()),
+        "Precursor Orbs - LPC": set(Orbs.locLPC_orbBundleTable.values()),
+        "Precursor Orbs - BS": set(Orbs.locBS_orbBundleTable.values()),
+        "Precursor Orbs - MP": set(Orbs.locMP_orbBundleTable.values()),
+        "Precursor Orbs - VC": set(Orbs.locVC_orbBundleTable.values()),
+        "Precursor Orbs - SC": set(Orbs.locSC_orbBundleTable.values()),
+        "Precursor Orbs - SM": set(Orbs.locSM_orbBundleTable.values()),
+        "Precursor Orbs - LT": set(Orbs.locLT_orbBundleTable.values()),
+        "Precursor Orbs - GMC": set(Orbs.locGMC_orbBundleTable.values()),
+        "Trades": {location_table[Cells.to_ap_id(k)] for k in
+                   {11, 12, 31, 32, 33, 96, 97, 98, 99, 13, 14, 34, 35, 100, 101}},
+        "'Free 7 Scout Flies' Power Cells": set(Cells.loc7SF_cellTable.values()),
+    }
+
+    # These functions and variables are Options-driven, keep them as instance variables here so that we don't clog up
+    # the seed generation routines with options checking. So we set these once, and then just use them as needed.
+    can_trade: Callable[[CollectionState, int, int | None], bool]
+    total_orbs: int = 2000
+    orb_bundle_item_name: str = ""
+    orb_bundle_size: int = 0
+    total_trade_orbs: int = 0
+    total_prog_orb_bundles: int = 0
+    total_trap_orb_bundles: int = 0
+    total_filler_orb_bundles: int = 0
+    total_power_cells: int = 101
+    total_prog_cells: int = 0
+    total_trap_cells: int = 0
+    total_filler_cells: int = 0
+    power_cell_thresholds: list[int] = []
+    trap_weights: tuple[list[str], list[int]] = ({}, {})
+
+    # Handles various options validation, rules enforcement, and caching of important information.
+    def generate_early(self) -> None:
+
+        # Cache the power cell threshold values for quicker reference.
+        self.power_cell_thresholds = []
+        self.power_cell_thresholds.append(self.options.fire_canyon_cell_count.value)
+        self.power_cell_thresholds.append(self.options.mountain_pass_cell_count.value)
+        self.power_cell_thresholds.append(self.options.lava_tube_cell_count.value)
+        self.power_cell_thresholds.append(100)  # The 100 Power Cell Door.
+
+        # Order the thresholds ascending and set the options values to the new order.
+        try:
+            if self.options.enable_ordered_cell_counts:
+                self.power_cell_thresholds.sort()
+                self.options.fire_canyon_cell_count.value = self.power_cell_thresholds[0]
+                self.options.mountain_pass_cell_count.value = self.power_cell_thresholds[1]
+                self.options.lava_tube_cell_count.value = self.power_cell_thresholds[2]
+        except IndexError:
+            pass  # Skip if not possible.
```

Comment:
should this log some kind of warning? I don't even see how this is possible.

Reply:
I think I was maybe overly defensive in my programming there.